src/roact.ts

```diff
@@ -0,0 +1,167 @@
+import * as vscode from "vscode"
+import { ApiClass, ApiPropertySecurity, getApiDump, UNCREATABLE_TAGS } from "./dump"
+
+const UNSCRIPTABLE_TAGS: Set<string> = new Set([
+    "Deprecated",
+    "Hidden",
+    "NotBrowsable",
+    "NotScriptable",
+])
+
+const isCreatableInstance = (klass: ApiClass) => {
+    const tags = klass.Tags
+    if (tags) {
+        for (const tag of tags) {
+            if (UNCREATABLE_TAGS.has(tag)) {
+                return false
+            }
+        }
+    }
+
+    return true
+}
+
+export class RoactCompletionProvider implements vscode.CompletionItemProvider {
+    instances: Promise<Map<string, ApiClass>>
+    creatableInstancesItems: Promise<vscode.CompletionItem[]>
+
+    constructor() {
+        this.instances = getApiDump().then(dump => {
+            const output = new Map()
+
+            for (const klass of dump.Classes) {
+                const klassData = {
+                    Description: klass.Description,
+                    Members: klass.Members.filter(member => {
+                        const tags = member.Tags
+                        if (tags !== undefined) {
+                            for (const tag of tags) {
+                                if (UNSCRIPTABLE_TAGS.has(tag)) {
+                                    return false
+                                }
+                            }
+                        }
+                        return true
+                    }),
```

Comment:
Turn this into a one line `every`.

Reply:
Didn't know about this before, good to know!